src/components/Calender/DateDisplay.tsx

```diff
@@ -1,76 +1,82 @@
+    const currentDate = useSelector((state) => state.calendarReducer.today);
+    const selectedDate = useSelector((state) => state.calendarReducer.selectedDate?.date);
 
-    function determineRenderDate(initial) {
-        let renderDate: Date = new Date(
-            initial.setDate(initial.getDate() - firstDay - 1),
-        );
+    function determineRenderDate(initial): () => Date {
+        let renderDate: Date = new Date(initial.setDate(initial.getDate() - firstDay));
 
         return updateDate;
+
         function updateDate() {
-            const returnDate = renderDate;
+            const returnDate: Date = new Date(renderDate);
             renderDate.setDate(renderDate.getDate() + 1);
             return returnDate;
         }
     }
 
-    const determineDate = determineRenderDate(initialDate);
-
-    const populateTasks = (date: Date): Object[] => {
-        const tasks = useSelector((state) => {
-            return state.taskReducer.tasks;
-        });
+    const determineDate: () => Date = determineRenderDate(initialDate);
 
-        date.setDate(date.getDate() - 1); //necessary to load the correct tasks into the correct tile
-        const filteredTasks = tasks.filter((task) => {
-            const tasksDate = new Date(task.date);
+    const populateTasks = (date: Date): TaskData[] => {
+        const filteredTasks = tasks.filter((task: TaskData) => {
+            const tasksDate: Date = new Date(task.date.replace(/-/g, '/'));
             return compareDates(tasksDate, date);
         });
 
         return filteredTasks;
     };
 
-    // create a function that does the math to find out with index on the array it needs to update
+    // console.log('date display rerendering', month, year);
 
-    const currentDate = useSelector((state) => {
-        return state.calendarReducer.today;
-    });
+    const onDateTilePressed = (date: Date, dateTasks: Object[]) => {
+        dispatch(selectDate(date, dateTasks));
+    };
 
     for (let j = 0; j < dateArray.length; j++) {
         dateArray[j] = new Array(7);
         for (let k = 0; k < dateArray[j].length; k++) {
-            const day: Date = determineDate();
+            const day: Date = React.useMemo(() => determineDate(), [determineDate]);
+            const data: TaskData[] = React.useMemo(() => populateTasks(day), []);
             const isToday: boolean = compareDates(day, currentDate);
-            const data: Object[] = populateTasks(new Date(day));
             const isCurrentMonth: boolean = day.getMonth() === month;
+            let isSelected = false;
+            if (compareDates(day, selectedDate)) {
+                isSelected = true;
+            }
 
             dateArray[j][k] = (
                 <DateTile
                     data={data}
-                    renderedDate={new Date(day)}
-                    key={j + j * (k + 1) + k}
+                    day={day}
+                    key={`${day.toDateString}${j}${k}`}
```

Comment:
better way to denote keys that are strings concatenated with numbers is to have a '-' in between them.